Firestore/Source/API/FIRQuery.mm

```diff
@@ -626,11 +627,12 @@ - (Bound)boundFromFieldValues:(NSArray<id> *)fieldValues isBefore:(BOOL)isBefore
                              path.CanonicalString());
       }
       DocumentKey key{path};
-      FreeNanopbMessage(google_firestore_v1_Value_fields, &fieldValue);
-      fieldValue = RefValue(self.firestore.databaseID, key);
+      components.values[idx] = RefValue(self.firestore.databaseID, key);
+    } else {
+      fieldValue.release();
+      components.values[idx] = *fieldValue;
```

Comment:
Optional nit: `release` returns a pointer, so this could be
```cpp
components.values[idx] = *fieldValue.release();
```

Reply:
I will fix this in the main PR.